### DISCUSSION TITLE
Unknown: Horovod has been shut down？

### DISCUSSION THREAD

**Discussion by Jiancong** [horovod/horovod#3076] [Q&A]
**Environment:**

1. Framework: Tensorflow
2. Framework version: 1.15.0
3. Horovod version: 0.21.3
4. MPI version: 4.0.0
5. CUDA version: V10.0.130
6. NCCL version:2408
7. Python version: 3.7
8. Spark / PySpark version: 
9. Ray version:
10. OS and version: debian buster/sid / ubuntu
11. GCC version: 7.5.0
12. CMake version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe erroneous behavior you're observing and steps to reproduce it.

I've tried distributed trained on 4 workers and each with 2 gpu cards and got following errors.

'''
mpirun -np 8 --allow-run-as-root  -H ${ip[0]}:2,${ip[1]}:2,${ip[2]}:2,${ip[3]}:2  python3 ./dssm.py --train_data=${train_data} --val_data=${train_data} --test_data=${eval_data} --infer_result=${infer_res} --ckpt_save_dir=${ckpt_save_dir} --servable_model_dir=${servable_model_dir} --task_type=${task_type} --num_epochs=${num_epochs} --gpu_slot=${gpu_slot} --max_steps=${max_steps} --batch_size=${batch_size} --log_steps=${log_steps} --learning_rate=${learning_rate} --embedding_size=${embedding_size}  --num_split=${num_split}
'''

 It reports following error message:

'''
2021/06/11 17:14:04 2021-06-11 17:14:02.698050: W tensorflow/core/framework/op_kernel.cc:1651] OP_REQUIRES failed at mpi_ops.cc:684 : Unknown: Horovod has been shut down. This was caused by an exception on one of the ranks or an attempt to allreduce, allgather or broadcast a tensor after one of the ranks finished execution. If the shutdown was caused by an exception, you should see the exception in the log before the first shutdown message.
2021/06/11 17:14:04 Traceback (most recent call last):
2021/06/11 17:14:04 2021-06-11 17:14:02.698164: W tensorflow/core/framework/op_kernel.cc:1651] OP_REQUIRES failed at mpi_ops.cc:684 : Unknown: Horovod has been shut down. This was caused by an exception on one of the ranks or an attempt to allreduce, allgather or broadcast a tensor after one of the ranks finished execution. If the shutdown was caused by an exception, you should see the exception in the log before the first shutdown message.
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/client/session.py", line 1350, in _run_fn
2021/06/11 17:14:04     run_metadata)
2021/06/11 17:14:04 	 [[{{node HorovodBroadcast_Variable_0}}]]
2021/06/11 17:14:04 	 [[{{node HorovodBroadcast_Variable_0}}]]
2021/06/11 17:14:04 
2021/06/11 17:14:04 Traceback (most recent call last):
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/platform/app.py", line 40, in run
2021/06/11 17:14:04     _run_main(main, args)
2021/06/11 17:14:04   File "./dssm.py", line 763, in main
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1195, in _train_model_default
2021/06/11 17:14:04     loss = self._train_model(input_fn, hooks, saving_listeners)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/training/monitored_session.py", line 1014, in __init__
2021/06/11 17:14:04     log_step_count_steps=log_step_count_steps) as mon_sess:
2021/06/11 17:14:04     self._sess = _RecoverableSession(self._coordinated_creator)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/training/monitored_session.py", line 1212, in _create_session
2021/06/11 17:14:04     hook.after_create_session(self.tf_sess, self.coord)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/client/session.py", line 956, in run
2021/06/11 17:14:04     feed_dict_tensor, options, run_metadata)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/client/session.py", line 1384, in _do_call
2021/06/11 17:14:04   (0) Unknown: Horovod has been shut down. This was caused by an exception on one of the ranks or an attempt to allreduce, allgather or broadcast a tensor after one of the ranks finished execution. If the shutdown was caused by an exception, you should see the exception in the log before the first shutdown message.
2021/06/11 17:14:04   (1) Unknown: Horovod has been shut down. This was caused by an exception on one of the ranks or an attempt to allreduce, allgather or broadcast a tensor after one of the ranks finished execution. If the shutdown was caused by an exception, you should see the exception in the log before the first shutdown message.
2021/06/11 17:14:04 0 derived errors ignored.
2021/06/11 17:14:04   File "./dssm.py", line 892, in <module>
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/absl/app.py", line 251, in _run_main
2021/06/11 17:14:04     _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_estimator/python/estimator/estimator.py", line 1161, in _train_model
2021/06/11 17:14:04     model.train(input_fn=train_input_fn, max_steps=FLAGS.max_steps // hvd.size(), hooks=[bcast_hook])
2021/06/11 17:14:04     saving_listeners)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/horovod/tensorflow/__init__.py", line 327, in begin
2021/06/11 17:14:04     stop_grace_period_secs=stop_grace_period_secs)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/training/monitored_session.py", line 584, in MonitoredTrainingSession
2021/06/11 17:14:04     return broadcast_variables(_global_variables(), root_rank)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/horovod/tensorflow/functions.py", line 42, in broadcast_group
2021/06/11 17:14:04     for var in variables])
2021/06/11 17:14:04   File "<string>", line 459, in horovod_broadcast
2021/06/11 17:14:04     attrs, op_def, compute_device)
2021/06/11 17:14:04   File "/usr/local/lib/python3.7/dist-packages/tensorflow_core/python/util/deprecation.py", line 507, in new_func 
'''

Could anyone show some hint?
Jiancong

**Comment by maxhgerlach** [comment]
It looks as if one of the ranks crashed. It would be helpful to look at the logs of all Horovod processes to find the cause. If you add an argument like `--tag-output` to `mpirun` it will be easier to discern the output.
